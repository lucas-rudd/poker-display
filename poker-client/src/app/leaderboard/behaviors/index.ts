import { withFetchPokerPlayers } from './withFetchPokerPlayers';
import { compose, lifecycle, withProps } from 'recompose';

const formatWinnings = (winnings) => {
    if(winnings >= 1000000000) {
        return `$${(winnings / 1000000000).toFixed(1)}B`;
    }
    if(winnings >= 1000000) {
        return `$${(winnings / 1000000).toFixed(1)}M`;
    }
    if(winnings >= 1000) {
        return `$${(winnings / 1000).toFixed(1)}K`
    }
    return `$${winnings.toFixed(1)}`
};

export const enhance = compose(
    withProps({
        pokerPlayers: [],
        fetching: false,
        error: {}
    }),
    lifecycle({
        componentWillMount() {
            this.setState({ fetching: true });
            withFetchPokerPlayers({ order: 'desc', sortField: 'winnings'}).then(pokerPlayers => {
                this.setState({ fetching: false});
                this.setState({ pokerPlayers });
            }).catch((error) => {
                this.setState({fetching: false});
                this.setState({ error })
            })
        },
        componentWillUpdate(_, nextState) {
            if(nextState.pokerPlayers) {
                nextState.pokerPlayers = nextState.pokerPlayers.map(({winnings, ...rest}) => {
                    return { winnings: formatWinnings(winnings), ...rest }
                })
            }
        }
    })
)