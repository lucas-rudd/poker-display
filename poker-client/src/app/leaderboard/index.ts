export * from './container';
export * from './poker-player';