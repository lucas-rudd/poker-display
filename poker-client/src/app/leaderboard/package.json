{
  "name": "@poker-client/leaderboard",
  "version": "1.0.0",
  "description": "The leaderboard component",
  "author": "Lucas Rudd",
  "private": true,
  "dependencies": {
    "@types/react": "^16.8.15",
    "axios": "^0.18.0",
    "country-data": "^0.0.31",
    "react": "16.8.6",
    "react-country-flag": "^1.0.2",
    "recompose": "^0.30.0",
    "styled-components": "^4.2.0"
  }
}
