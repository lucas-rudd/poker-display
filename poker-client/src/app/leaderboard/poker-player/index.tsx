import React from 'react';
import ReactCountryFlag from 'react-country-flag';
import CountryData from 'country-data'
import { PokerPlayerName } from './styles';

const hasCountry = (country) => {
    return CountryData.countries.hasOwnProperty(country);
}

export const PokerPlayer: React.FC<{id, firstName, lastName, winnings, image, country}> = ({ firstName, lastName, winnings, image, country='' }) => {
    const countryCode = country.toUpperCase();
    return (
        <React.Fragment>
            <th scope="row">
                <div style={{display: 'flex', alignItems: 'center'}}>
                    <img height='48px' width='48px' src={image || `${process.env.PUBLIC_URL}/no_image_available.png`}/>
                    <PokerPlayerName>{firstName} {lastName}</PokerPlayerName>
                </div>
            </th>
            <td>{winnings}</td>
            <td>
                <div style={{display: 'flex', alignItems: 'center'}}>
                    <span style={{fontSize: '150%'}}>
                          {hasCountry(country) ? CountryData.countries[countryCode].emoji : '' }
                    </span>
                    <div style={{paddingLeft: '10px'}}>
                        {hasCountry(country) ? CountryData.countries[countryCode].name : ''}
                    </div>
                </div>

            </td>
        </React.Fragment>
    )
}