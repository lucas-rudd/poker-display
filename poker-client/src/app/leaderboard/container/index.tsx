import React from 'react';
import { PokerPlayer } from '../poker-player';
import { PokerPlayerTable, PokerPlayerTableCaption } from './styles';
import logo from '../../logo.svg';
import { enhance } from '../behaviors';

const renderPokerPlayers = (pokerPlayers) => {
    return pokerPlayers.map(({_id, firstName, lastName, image, winnings, country}) => {
        return (
            <tr className="poker-player-table-row">
                <PokerPlayer id={_id} firstName={firstName} lastName={lastName} image={image} winnings={winnings} country={country}/>
            </tr>
        );
    })
}

const PokerPlayerContainer: React.FC<{pokerPlayers, fetching?, error?}> = enhance(({fetching = false, error = {}, pokerPlayers}) => {
    const tableBody = fetching ? (<img src={logo} className="app-logo" alt="logo" />) : renderPokerPlayers(pokerPlayers);
        return (
            <div className="scroller">
                <PokerPlayerTable className="poker-player-table">
                    <PokerPlayerTableCaption>ALL-TIME TOURNAMENT EARNINGS</PokerPlayerTableCaption>
                    <thead className="poker-player-header">
                        <tr className="poker-player-table-row">
                            <th scope="col">Player</th>
                            <th scope="col">Winnings</th>
                            <th scope="col">Native Of</th>
                        </tr>
                    </thead>
                    <tbody>
                        {tableBody}
                    </tbody>
                </PokerPlayerTable>
            </div>
        );
    });

export const PokerPlayerLeaderboard = PokerPlayerContainer;
