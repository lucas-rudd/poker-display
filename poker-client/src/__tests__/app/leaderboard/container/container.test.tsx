import React from 'react';
import renderer from 'react-test-renderer';
import { PokerPlayerContainer } from '../../../../app/leaderboard';

const pokerPlayers = [{
    firstName: 'firstName',
    lastName: 'lastName',
    winnings: 100,
    image: 'base64EncodedImage',
    country: 'us'
}];

it('renders without crashing', () => {
    const tree = renderer.create(<PokerPlayerContainer fetching={true} error={{}} pokerPlayers={[]} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('renders table when pokerPlayer data and not fetching', () => {
    const tree = renderer.create(<PokerPlayerContainer fetching={false} error={{}} pokerPlayers={pokerPlayers} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})