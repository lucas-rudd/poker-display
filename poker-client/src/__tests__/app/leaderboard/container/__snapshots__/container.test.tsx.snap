// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table when pokerPlayer data and not fetching 1`] = `
<div
  className="scroller"
>
  <table
    className="poker-player-table sc-bwzfXH hCaaRj"
  >
    <caption
      className="sc-htpNat gWngkR"
    >
      ALL-TIME TOURNAMENT EARNINGS
    </caption>
    <thead
      className="poker-player-header"
    >
      <tr
        className="poker-player-table-row"
      >
        <th
          scope="col"
        >
          Player
        </th>
        <th
          scope="col"
        >
          Winnings
        </th>
        <th
          scope="col"
        >
          Native Of
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="poker-player-table-row"
      >
        <th
          scope="row"
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
              }
            }
          >
            <img
              height="48px"
              src="base64EncodedImage"
              width="48px"
            />
            <div
              className="sc-bdVaJa hniAuI"
            >
              firstName
               
              lastName
            </div>
          </div>
        </th>
        <td>
          100
        </td>
        <td>
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
              }
            }
          >
            <span
              style={
                Object {
                  "fontSize": "150%",
                }
              }
            >
              
            </span>
            <div
              style={
                Object {
                  "paddingLeft": "10px",
                }
              }
            >
              
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="scroller"
>
  <table
    className="poker-player-table sc-bwzfXH hCaaRj"
  >
    <caption
      className="sc-htpNat gWngkR"
    >
      ALL-TIME TOURNAMENT EARNINGS
    </caption>
    <thead
      className="poker-player-header"
    >
      <tr
        className="poker-player-table-row"
      >
        <th
          scope="col"
        >
          Player
        </th>
        <th
          scope="col"
        >
          Winnings
        </th>
        <th
          scope="col"
        >
          Native Of
        </th>
      </tr>
    </thead>
    <tbody>
      <td
        colSpan={3}
      >
        <img
          alt="logo"
          className="app-logo"
          src="logo.svg"
        />
      </td>
    </tbody>
  </table>
</div>
`;
